graph/topo/paton_cycles.go

```diff
@@ -0,0 +1,86 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package topo
+
+import (
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/internal/linear"
+	"gonum.org/v1/gonum/graph/internal/set"
+)
+
+// UndirectedCyclesIn returns the set of cycle bases in the graph g.
```

Comment:
For me the basis would be the set itself, and the cycles would then be called basis cycles. I would maybe change the comment to
```
UndirectedCyclesIn returns a set of cycles that forms a cycle basis in the graph g.
Any cycle in g can be constructed as a symmetric difference of its elements.
```